src/MongoDB.Driver.Core/MongoNodeIsRecoveringException.cs

```diff
@@ -71,5 +71,25 @@ protected MongoNodeIsRecoveringException(SerializationInfo info, StreamingContex
         {
         }
 #endif
+
+        /// <summary>
+        /// TODO: through ctor?
+        /// </summary>
+        /// <returns>TODO</returns>
+        public bool IsShutdownError
```

Comment:
sometimes we need slightly narrower flag, so added it here similar to what we had in MongoConnectionException with `ContainsTimeoutException`